food-delivery/packages/activities/services.ts

```diff
@@ -0,0 +1,31 @@
+import { ApplicationFailure } from '@temporalio/activity'
+
+export const notificationService = {
+  sendNotification({ type, message }: { type: string; message: string }) {
+    if (Math.random() < 0.7) {
+      throw new Error(`Failed to send ${type} notification. Unable to reach notification service.`)
+    }
+
+    console.log(`Sent ${type} notification: ${message}`)
+  },
+}
+
+export const paymentService = {
+  charge(amount: number) {
```

Comment:
Use cents to avoid floating point arithmetic

Reply:
Everything is in whole numbers for simplicity. Doesn't seem like this is worth the additional code?

```
diff --git a/food-delivery/packages/activities/index.ts b/food-delivery/packages/activities/index.ts
index 0b51953..ba531c3 100644
--- a/food-delivery/packages/activities/index.ts
+++ b/food-delivery/packages/activities/index.ts
@@ -6,9 +6,9 @@ export async function sendPushNotification(message: string): Promise<void> {
 }
 
 export async function refundOrder(product: Product): Promise<void> {
-  paymentService.refund(product.price)
+  paymentService.refund({ cents: product.price * 100 })
 }
 
 export async function chargeCustomer(product: Product): Promise<void> {
-  paymentService.charge(product.price)
+  paymentService.charge({ cents: product.price * 100 })
 }
diff --git a/food-delivery/packages/activities/services.ts b/food-delivery/packages/activities/services.ts
index 3bfcc41..4a7d535 100644
--- a/food-delivery/packages/activities/services.ts
+++ b/food-delivery/packages/activities/services.ts
@@ -11,25 +11,25 @@ export const notificationService = {
 }
 
 export const paymentService = {
-  charge(amount: number) {
+  charge({ cents }: { cents: number }) {
     // In a real app, we would pass an idempotency token to the downstream service
     const _idempotencyToken = Context.current().info.workflowExecution.workflowId
-    if (amount >= 35) {
+    if (cents >= 3500) {
       throw ApplicationFailure.create({ nonRetryable: true, message: 'Card declined: insufficient funds' })
     }
     if (Math.random() < 0.7) {
       throw new Error('Failed to charge. Unable to reach payment service.')
     }
-    console.log(`Refunded $${amount}`)
+    console.log(`Refunded $${cents / 100}`)
   },
 
-  refund(amount: number) {
+  refund(cents: number) {
     // In a real app, we would pass an idempotency token to the downstream service
     const _idempotencyToken = Context.current().info.workflowExecution.workflowId
     if (Math.random() < 0.7) {
       throw new Error('Failed to refund. Unable to reach payment service.')
     }
 
-    console.log(`Refunded $${amount}`)
+    console.log(`Refunded $${cents / 100}`)
   },
 }
```

Reply:
Use cents everywhere, format the amounts in the UI.
If you want references to where and why to use this practice, LMK.